client/src/pages/PostSolution.tsx

```diff
@@ -44,23 +46,24 @@ const PostSolution = () => {
   const handleSubmit = (e: FormEvent<HTMLFormElement>) => {
     e.preventDefault();
     // TODO: Form validation
-    console.log(formData);
-    console.log(challengeId);
+    const data = new FormData();
+    data.append("userId", `${userId}`);
+    data.append("challengeId", `${challengeId}`);
+
+    data.append("title", formData.title);
+
+    data.append("description", formData.description);
+    data.append("diagram", formData.diagram);
 
     fetch(`/api/solutions`, {
-      method: "POST", // or 'PUT'
-      headers: {
-        "Content-Type": "application/json",
-      },
```

Comment:
Do we not need `headers`?

Reply:
We're uploading the file in multiple parts which we cannot do with `application/json`. So we switched over to `multipart/form-data`. The browser has an api called FormData (which we now use) that automatically handles this process, which includes adding the headers.